cmd/commands/daemon/command.go

```diff
@@ -32,11 +31,15 @@ func NewCommand() *cli.Command {
 		Usage:     "Starts Mysterium Tequilapi service",
 		ArgsUsage: " ",
 		Action: func(ctx *cli.Context) error {
+			errorChannel := make(chan error, 1)
 			if err := di.Bootstrap(cmd.ParseFlagsNode(ctx)); err != nil {
 				return err
 			}
-			cmd.RegisterSignalCallback(utils.SoftKiller(di.Shutdown))
-			return di.Node.Wait()
+			go func() { errorChannel <- di.Node.Wait() }()
+
+			cmd.RegisterSignalCallback(func() { errorChannel <- nil })
```

Comment:
I see a channel of size 1 here, and two writes (from Node.Wait goroutine, and signaling goroutine), only one read. Another thing - when signal is received and Action exits (return <-errorChannel), what happens with Node which is still running? is it stopped somewhere else? doesn't sound right.

Reply:
A node will be stopped by the `di.Shutdown()` in the `After` section, a bit lower.

Reply:
Buffered channel is used here to not block goroutine and release it without waiting of reading.

Reply:
My hint was maybe a channel with size 2 should be more appropriate? So each goroutine can put its result independenly whenever another party read previous value or not

Reply:
Make sense, especially not only in this case but in other, with more waiters. 
Changed it in all places.